src/spyglass/decoding/v1/sorted_spikes.py

```diff
@@ -0,0 +1,367 @@
+
+    def load_results(self):
+        return SortedSpikesDetector.load_results(self.fetch1("results_path"))
+
+    @staticmethod
+    def load_environments(key):
+        model_params = (
+            DecodingParameters
+            & {"decoding_param_name": key["decoding_param_name"]}
+        ).fetch1()
+        decoding_params, decoding_kwargs = (
+            model_params["decoding_params"],
+            model_params["decoding_kwargs"],
+        )
+
+        (
+            position_info,
+            position_variable_names,
+        ) = SortedSpikesDecodingV1.load_position_info(key)
+        classifier = SortedSpikesDetector(**decoding_params)
+
+        classifier.initialize_environments(
+            position=position_info[position_variable_names].to_numpy(),
+            environment_labels=decoding_kwargs.get("environment_labels", None),
+        )
+
+        return classifier.environments
+
+    @staticmethod
+    def load_position_info(key):
+        position_group_key = {"position_group_name": key["position_group_name"]}
+        position_variable_names = (PositionGroup & position_group_key).fetch1(
+            "position_variables"
+        )
+
+        position_info = []
+        for pos_merge_id in (PositionGroup.Position & position_group_key).fetch(
+            "pos_merge_id"
+        ):
+            position_info.append(
+                (PositionOutput & {"merge_id": pos_merge_id}).fetch1_dataframe()
+            )
+        position_info = pd.concat(position_info, axis=0).dropna()
+
+        return position_info, position_variable_names
+
+    @staticmethod
+    def _get_interval_range(key):
+        encoding_interval = (
+            IntervalList
+            & {
+                "nwb_file_name": key["nwb_file_name"],
+                "interval_list_name": key["encoding_interval"],
+            }
+        ).fetch1("valid_times")
+
+        decoding_interval = (
+            IntervalList
+            & {
+                "nwb_file_name": key["nwb_file_name"],
+                "interval_list_name": key["decoding_interval"],
+            }
+        ).fetch1("valid_times")
+
+        return (
+            min(
+                np.asarray(encoding_interval).min(),
+                np.asarray(decoding_interval).min(),
+            ),
+            max(
+                np.asarray(encoding_interval).max(),
+                np.asarray(decoding_interval).max(),
+            ),
+        )
+
+    @staticmethod
+    def load_spike_data(key, filter_by_interval=True):
+        merge_ids = (
+            (
+                SortedSpikesGroup.SortGroup
+                & {"sorted_spikes_group_name": key["sorted_spikes_group_name"]}
+            )
+        ).fetch("spikesorting_merge_id")
+
+        spike_times = [
+            SpikeSortingOutput.fetch_nwb({"merge_id": merge_id})[0][
+                "object_id"
+            ]["spike_times"].to_list()
+            for merge_id in merge_ids
+        ]
```

Comment:
```suggestion
       nwb_files = [
            SpikeSortingOutput.fetch_nwb({"merge_id": merge_id})[0]
            for merge_id in merge_ids
        ]
        spike_times = [
            nwb_file["object_id"]["spike_times"].to_list()  # v1 spikesorting
            if "object_id" in nwb_file
            else nwb_file["units"]["spike_times"].to_list()  # v0 spikesorting
            if "units" in nwb_file
            else []  # no units in spikesorting
            for nwb_file in nwb_files
        ]
```

Change this to accept formatting from multiple spikesorting pipeline sources and handle groups that have no units in the selected curation